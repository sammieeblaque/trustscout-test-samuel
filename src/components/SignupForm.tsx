import React, { Component, SyntheticEvent } from "react";
import { Button, Card, Form, Alert } from "react-bootstrap";

import * as routes from "../routes";
import { auth, db } from "../firebase/";

interface IinitialState {
  username: string;
  email: string;
  password: string;
  error: any;
  showAlert: boolean;
}

interface IUserProps {
  history?: any;
}

class SignupForm extends Component<IUserProps> {
  state: IinitialState = {
    username: "",
    email: "",
    password: "",
    error: null,
    showAlert: false,
  };

  onSubmit = async (event: SyntheticEvent) => {
    event.preventDefault();
    const { username, password, email } = this.state;
    const { history } = this.props;
    try {
      const authUser = await auth.doCreateUserWithEmailAndPassword(
        email,
        password
      );
      await db.doCreateUser(authUser.user?.uid ?? "", username, email);
      await this.setState({ ...this.state });
      // console.log(this.state);
      history.push(routes.HOME);
    } catch (error) {
      this.setState({ error: error, showAlert: true });
    }
  };

  render() {
    const { username, password, email, showAlert, error } = this.state;
    return (
      <>
        <div data-testid="signup">
          <Card className="no__border left__pad">
            <Card.Body className="w__90">
              <h2 className="text-center mb-4">Sign Up</h2>
              {showAlert && (
                <Alert
                  variant="danger"
                  onClose={() => this.setState({ showAlert: false })}
                  dismissible
                >
                  {error.message}
                </Alert>
              )}
              <Form onSubmit={this.onSubmit}>
                <Form.Group id="name">
                  <Form.Label>Name</Form.Label>
                  <Form.Control
                    type="text"
                    placeholder="John Doe"
                    value={username}
                    onChange={(e) =>
                      this.setState({ username: e.target.value })
                    }
                    required
                  />
                </Form.Group>
                <Form.Group id="email">
                  <Form.Label>Email</Form.Label>
                  <Form.Control
                    placeholder="john@xyz.com"
                    value={email}
                    onChange={(e) => this.setState({ email: e.target.value })}
                    required
                  />
                </Form.Group>
                <Form.Group id="password">
                  <Form.Label>Password</Form.Label>
                  <Form.Control
                    type="password"
                    value={password}
                    onChange={(e) =>
                      this.setState({ password: e.target.value })
                    }
                    required
                  />
                </Form.Group>
                <Button className="w-100 button__color" type="submit">
                  Sign Up
                </Button>
                <div className="text-center">
                  <p>
                    Already have an account?{" "}
                    <a href={routes.LANDING_PAGE}>Sign In</a>
                  </p>
                </div>
              </Form>
            </Card.Body>
          </Card>
        </div>
      </>
    );
  }
}

const SignUpLink = () => (
  <p>
    Don't have an account? <a href={routes.SIGN_UP}>Sign Up</a>
  </p>
);

export { SignupForm, SignUpLink };
