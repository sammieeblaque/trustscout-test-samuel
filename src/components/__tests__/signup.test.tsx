import { render, cleanup, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { SignupForm } from "../SignupForm";

afterEach(() => {
  cleanup();
});

describe("to run tests", () => {
  test("should render SignUp", () => {
    let history;
    render(<SignupForm history={history} />);
    const signupElement = screen.getByTestId("signup");
    expect(signupElement).toBeInTheDocument();
  });
});

test("Matches Signup Snapshot", () => {
  let history;
  const tree = renderer.create(<SignupForm history={history} />).toJSON();
  expect(tree).toMatchSnapshot();
});
