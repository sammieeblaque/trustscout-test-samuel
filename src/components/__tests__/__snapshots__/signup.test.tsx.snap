// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Signup Snapshot 1`] = `
<div
  data-testid="signup"
>
  <div
    className="no__border left__pad card"
  >
    <div
      className="w__90 card-body"
    >
      <h2
        className="text-center mb-4"
      >
        Sign Up
      </h2>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
          id="name"
        >
          <label
            className="form-label"
          >
            Name
          </label>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="John Doe"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
          id="email"
        >
          <label
            className="form-label"
          >
            Email
          </label>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="john@xyz.com"
            required={true}
            value=""
          />
        </div>
        <div
          className="form-group"
          id="password"
        >
          <label
            className="form-label"
          >
            Password
          </label>
          <input
            className="form-control"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className="w-100 button__color btn btn-primary"
          disabled={false}
          type="submit"
        >
          Sign Up
        </button>
        <div
          className="text-center"
        >
          <p>
            Already have an account?
             
            <a
              href="/"
            >
              Sign In
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Matches Snapshot 1`] = `
<div
  className="no__border left__pad card"
>
  <div
    className="w__90 card-body"
  >
    <h2
      className="text-center mb-4"
    >
      Sign Up
    </h2>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
        id="name"
      >
        <label
          className="form-label"
        >
          Name
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="John Doe"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
        id="email"
      >
        <label
          className="form-label"
        >
          Email
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="john@xyz.com"
          required={true}
          value=""
        />
      </div>
      <div
        className="form-group"
        id="password"
      >
        <label
          className="form-label"
        >
          Password
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="w-100 button__color btn btn-primary"
        disabled={false}
        type="submit"
      >
        Sign Up
      </button>
      <div
        className="text-center"
      >
        <p>
          Already have an account?
           
          <a
            href="/"
          >
            Sign In
          </a>
        </p>
      </div>
    </form>
  </div>
</div>
`;
